clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -503,6 +505,30 @@ public void testInterceptorConstructorClose() {
         }
     }
 
+    @Test
+    public void testInterceptorConstructorConfigurationWithExceptionShouldCloseRemainingInstances() {
```

Comment:
I like this test. Could we also get a matching one for producer interceptors, and something analogous for the `AbstractConfig` class?